src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx

```diff
@@ -0,0 +1,199 @@
+import { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { PageTitles } from 'common/constants';
+import { productVersionEntityAttributes } from 'common/productVersionEntityAttributes';
+
+import { IFieldValues, useForm } from 'hooks/useForm';
+import { useServiceContainer } from 'hooks/useServiceContainer';
+import { useTitle } from 'hooks/useTitle';
+
+import { ContentBox } from 'components/ContentBox/ContentBox';
+import { PageLayout } from 'components/PageLayout/PageLayout';
+import { ServiceContainerCreatingUpdating } from 'components/ServiceContainers/ServiceContainerCreatingUpdating';
+
+import * as productVersionApi from 'services/productVersionApi';
+
+import { validateProductVersionName } from 'utils/formValidationHelpers';
+import { createSafePatch } from 'utils/patchHelper';
+import { generatePageTitle } from 'utils/titleHelper';
+
+const formConfig = {
+  version: {
+    isRequired: true,
+
+    validators: [
+      {
+        validator: validateProductVersionName,
+        errorMessage: 'The version should consist of two numeric parts separated by a dot (e.g. 1.0).',
+      },
+    ],
+  },
+  'attributes.brewTagPrefix': {
+    isRequired: true,
+  },
+};
+
+interface IProductVersionCreateEditPageProps {
+  isEditPage?: boolean;
+}
+
+export const ProductVersionCreateEditPage = ({ isEditPage = false }: IProductVersionCreateEditPageProps) => {
+  const { productId } = useParams();
+  const { productVersionId } = useParams();
+  const navigate = useNavigate();
+
+  // create page
+  const serviceContainerCreatePage = useServiceContainer(productVersionApi.createProductVersion);
+
+  // edit page - get method
+  const serviceContainerEditPageGet = useServiceContainer(productVersionApi.getProductVersion);
+  const serviceContainerEditPageGetRunner = serviceContainerEditPageGet.run;
+
+  // edit page - patch method
+  const serviceContainerEditPagePatch = useServiceContainer(productVersionApi.patchProductVersion);
+
+  const { register, setFieldValues, getFieldState, getFieldErrors, handleSubmit, isSubmitDisabled } = useForm();
+
+  useTitle(
+    generatePageTitle({
+      pageType: isEditPage ? 'Edit' : 'Create',
+      serviceContainer: serviceContainerEditPageGet,
+      firstLevelEntity: 'Product',
+      nestedEntity: 'Version',
+      entityName: `${serviceContainerEditPageGet.data?.version} ${PageTitles.delimiterSymbol} ${serviceContainerEditPageGet.data?.product.name}`,
+    })
+  );
+
+  const submitCreate = (data: IFieldValues) => {
+    return serviceContainerCreatePage
+      .run({
+        serviceData: { data: { ...data, product: { id: productId } } },
+      })
+      .then((response: any) => {
+        const newProductVersionId = response?.data?.id;
+        if (!newProductVersionId) {
+          throw new Error(`Invalid newProductVersionId coming from Orch POST response: ${newProductVersionId}`);
```

Comment:
Do we need to consider using `uiLogger` as this is a pnc side error that we need to take care of?

Reply:
This issue will be considered in different Jira (NCL-8169).